app/src/main/java/org/p2p/wallet/newsend/model/CalculationMode.kt

```diff
@@ -51,16 +53,17 @@ class CalculationMode(
     private var usdAmount: BigDecimal = BigDecimal.ZERO
 
     fun updateToken(newToken: Token.Active) {
-        if (::token.isInitialized && newToken.mintAddress == this.token.mintAddress) return
-        this.token = newToken
+        if (::token.isInitialized && newToken.mintAddress == this.token.mintAddress) {
```

Comment:
when coming back from other screens in Send flow, updateToken was called, but didn't call updateLabels()